src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -503,17 +503,29 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 
 # DNS
 register_cli_argument('network dns', 'location', help=argparse.SUPPRESS, default='global')
-register_cli_argument('network dns zone', 'zone_name', name_arg_type)
 register_cli_argument('network dns', 'record_set_name', name_arg_type, help='The name of the RecordSet, relative to the name of the zone.')
 register_cli_argument('network dns', 'relative_record_set_name', name_arg_type, help='The name of the RecordSet, relative to the name of the zone.')
-register_cli_argument('network dns', 'zone_name', help='The name of the zone without a terminating dot.')
-register_cli_argument('network dns record-set create', 'record_set_type', options_list=('--type',), **enum_choice_list(RecordType))
-register_cli_argument('network dns record-set create', 'ttl', default=3600)
-register_cli_argument('network dns', 'record_type', options_list=('--type',), **enum_choice_list(RecordType))
-register_cli_argument('network dns record', 'record_set_name', options_list=('--record-set-name',))
-register_cli_argument('network dns record txt add', 'value', nargs='+')
-register_cli_argument('network dns record txt remove', 'value', nargs='+')
+register_cli_argument('network dns', 'zone_name', options_list=('--zone-name', '-z'), help='The name of the zone without a terminating dot.')
+register_cli_argument('network dns', 'metadata', nargs='+', help='Metadata in space-separated key=value pairs. This overwrites any existing metadata.', validator=validate_metadata)
+
+modified_record_type = [x.value for x in RecordType if x.value != 'SOA']
+register_cli_argument('network dns', 'record_type', options_list=('--type', '-t'), **enum_choice_list(modified_record_type))
+register_cli_argument('network dns', 'location', help=argparse.SUPPRESS, default='global')
+
+register_cli_argument('network dns zone', 'zone_name', name_arg_type)
+
 register_cli_argument('network dns zone import', 'file_name', help='Path to the DNS zone file to import')
 register_cli_argument('network dns zone export', 'file_name', help='Path to the DNS zone file to save')
-register_cli_argument('network dns', 'location', help=argparse.SUPPRESS, default='global')
 register_cli_argument('network dns zone update', 'if_none_match', ignore_type)
+
+register_cli_argument('network dns record', 'record_set_name', options_list=('--record-set-name',))
+
+register_cli_argument('network dns record txt add', 'value', nargs='+')
+register_cli_argument('network dns record txt remove', 'value', nargs='+')
+
+register_cli_argument('network dns record-set create', 'record_set_type', options_list=('--type', '-t'), help='The type of DNS records in the record set.', **enum_choice_list(modified_record_type))
```

Comment:
You can still reference an arg_type and then override a piece of it, so this is a great suggestion. 👍